src/map/npc.c

```diff
@@ -5028,7 +5028,7 @@ static const char *npc_parse_mapflag(const char *w1, const char *w2, const char
 		struct map_zone_data *zone;
 		if (state != 0) {
 			if (w4 != NULL && sscanf(w4, "%d", &state) == 1)
-				map->list[m].flag.battleground = (state != 0) ? 1 : 0;
+				map->list[m].flag.battleground = cap_value(state, 1, 2);
```

Comment:
may be better cap from 0? ``cap_value(state, 0, 2)``
because with your code if put parameter as 0, it will be used as 1, i think this is wrong.

Reply:
```
	else if (strcmpi(w3, "battleground") == 0) {
		struct map_zone_data *zone;
		if (state != 0) {
			if (w4 != NULL && sscanf(w4, "%d", &state) == 1)
				map->list[m].flag.battleground = (state != 0) ? 1 : 0;
			else
				map->list[m].flag.battleground = 1; // Default value
		} else {
			map->list[m].flag.battleground = 0;
		}
```
wait, you mean we can use
```
prontera	mapflag	battleground	0
--same as--
//prontera	mapflag	battleground // not declare it
```
to turn off the mapflag ?? then what's the point of adding a mapflag line then ?

Reply:
yes i mean this, but i not saw check before. it blocks ``0`` already.
this can be need if don't want remove map flag but want disable it.

hm, but may be add value validation check? and if it not valid, show error message?
because old and new code silently accept any values...

Reply:
I really don't see the point of using value 0 
there are already *removemapflag script command that can remove a mapflag

just changing the value 1 into 0 doesn't mean anything, but doing it doesn't cost me my arm or leg, so I'll do it